enhancements/api-review/custom-tags-aws.md

```diff
@@ -79,65 +101,373 @@ The userTags field is intended to be set at install time and is considered immut
+
+   Previous tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`
+
+   Action = Update existing tag with value from `.spec.platformSpec.aws.resourceTags`.
+
+   Final tag set to AWS resource = `key_infra1 = value_infra1`
+
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+2) User deletes the user-defined tag from `.spec.platformSpec.aws.resourceTags`
+   The user-defined tag which is removed from spec, will not be reconciled or managed by operators.
+
+   User can update user-defined tag key:value using external tool. The user-defined tag will not be overwritten.
+
+   For example,
+
+   Existing tag for AWS resource = `key_infra1 = value1`
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has no user-defined tag with key `key_infra1`
+
+   Action = No change in tags for the AWS resource.
+
+   Final tag set to AWS resource = `key_infra1 = value1`
+
+   Event action = No event
+
+3) User sets user-defined tag created using external tools to empty string for deleting for AWS resource.
+
+   There is no validation check involved for creator tool of user-defined tag. Any user-defined tag listed by user in `.spec.platformSpec.aws.resourceTags` is considered for create/update/delete accordingly.
+
+4) Any user-defined tag set using `.spec.platformSpec.aws.resourceTags` in `Infrastructure.config.openshift.io/v1` type will affect all AWS resources managed.
+
+5) User-defined tags are not synced from `.spec.platformSpec.aws.resourceTags` to `.status.platformStatus.aws.resourceTags` for the following reasons.
+
+- User-defined tags in `.spec.platformSpec.aws.resourceTags` can be "removed without delete", "delete", "update".
+  `.spec.platformSpec.aws.resourceTags` to `.status.platformStatus.aws.resourceTags` sync is not required,
+  as there will be no versioning of the user-defined tags required to override the user-defined tags in Infrastructure CR.
+  Instead, the user-defined tags (if supported in resource operator spec field, e.g: machine) can override the user-defined tags in Infrastructure CR.
+
+- As `.spec.platformSpec.aws.resourceTags` has the actual values. Sync to `.status.platformStatus.aws.resourceTags` was required in earlier design to identify if the
+  tag on the resource was created using Infrastructure CR or external tool. Identifying the creator tool was done to restrict user from editing the same user-tag kv pair using multiple tools.
+  This inherently poses many scenarios of conflict which will result in user-defined tag kv pair being inconsistent across cluster when applied using Infrastructure CR. Hence, user will be confused which tool to be used to update tag.
+
+6) User applies user-defined tag using `.spec.platformSpec.aws.resourceTags`. Later, user modifies the user-defined tag for AWS resource using external tool.
```

Comment:
Let's make sure that the term "user-defined tag" is only being used for setting the tags on the kube resource.
```suggestion
6) User applies user-defined tag using `.spec.platformSpec.aws.resourceTags`. Later, user modifies the tag on the AWS resource using external tool.
```